dom0-updates/qfile-dom0-unpacker.c

```diff
@@ -90,7 +92,28 @@ int main(int argc, char ** argv)
 		files_limit = atoll(var);
 
 	set_size_limit(bytes_limit, files_limit);
-	if (argc > 3 && strcmp(argv[3],"-v")==0)
-		set_verbose(1);
+    for (i = 3; i < argc; i++) {
+        if (strcmp(argv[i], "-v") == 0) {
+            set_verbose(1);
+        } else if (strcmp(argv[i], "-w") == 0) {
+            if (i+1 < argc && argv[i+1][0] != '-') {
+                unsigned long space_margin;
+
+                space_margin = strtoul(argv[i+1], &var, 0);
+                if (*argv[i+1] == '\0' || *var != '\0' ||
+                        (space_margin == ULONG_MAX && errno == ERANGE)) {
```

Comment:
```suggestion
                errno = 0;
                space_margin = strtoul(argv[i+1], &var, 0);
                if (*var != '\0' || errno || *argv[i+1] < '0' || *argv[i+1] > '9') {
```

This ensures we don’t get a stale `errno` value, and checks for bogus leading whitespace.